### PR TITLE
Improve icon components

### PR DESCRIPTION
**Pull request by kanta1207**
## Overviews of implementation

- Recreated icon components to make it more reusable.
- Created new "icon" directory under the component directory, because it doesn't belong to "ui" directory.
- Add note to share how to create new icon components

## Review points

- [ ] Each of icon components are reusable enough?
- [ ] Was it right decision to create new "icon" directory under "components" directory, separated icons from "ui" directory?
- [ ] Does the directory structure in "icon" directory make sense?

### CODE REVIEW

**Review by uskayyyyy — COMMENTED** [review]
Thank you for the update! Please consider the following suggestions.

- [ ] I think SVG files should be stored separately as `.svg` image files under `public/images/icons/` or `src/assets/images/icons/` since they are IMAGES. It also benefits us to enable the preview of SVG files in your IDE. To style an SVG image, add `fill="currentColor"` to the <svg> tag, and style with Tailwind CSS class names after importing it.
mySvg.svg
```xml
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32" fill="currentColor">
  <path
    d="M11.56 13.3473C11.6269 13.5484... [rest of path data]"
  />
</svg>
```
myIconComponent.tsx
```tsx
import MySVG from './path/to/mySvg.svg';
:
return (
	<AppleImage className='text-blue-500' />
)
```
You also need to install `@svgr/webpack` and add the following configuration to the `next.config.js`.
```javacript
/** @type {import('next').NextConfig} */
const nextConfig = {
  :
  webpack(config) {
    config.module.rules.push({
      test: /\.svg$/,
      use: ['@svgr/webpack'],
    });
    return config;
  },
};

module.exports = nextConfig;
```
For more information, see: https://dev.to/dolearning/importing-svgs-to-next-js-nna

- [ ]

**Comment by uskayyyyy** [comment]
> # Review points
> - Each of icon components are reusable enough?
> - Was it right decision to create new "icon" directory under "components" directory, separated icons from "ui" directory?
> - Does the directory structure in "icon" directory make sense?

I think these are all good for now.

File: src/components/ui/BottomTabIconLink.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  isSelected: boolean;
+}
+
+export const BottomTabIconLink: React.FC<Props> = ({ name, isSelected }) => {
+  // Path and link title based on "mainRouteConfig" variable
+  const path = mainRouteConfig[name].path;
+  const title = mainRouteConfig[name].title;
+
+  // Icon component based on "isSelected" and "name" variable
+  const iconName: IconName = isSelected
```
Comment (kotaaaa) [inline src/components/ui/BottomTabIconLink.tsx:21]: Something looks like this seems better.

```
const iconName: IconName = {
  groups: isSelected ? 'home_on' : 'home_off',
  foods: isSelected ? 'food_on' : 'food_off',
  default: isSelected ? 'profile_on' : 'profile_off'
};

const iconName = iconNames[name] || iconNames.default;

```

or 


```
const iconName: IconName;
switch (name) {
  case 'groups':
    iconName = isSelected ? 'home_on' : 'home_off';
    break;
  case 'foods':
    iconName = isSelected ? 'food_on' : 'food_off';
    break;
  default:
    iconName = isSelected ? 'profile_on' : 'profile_off';
}

```

**Comment by kanta1207** [comment]
The check by PR Workflow was failing because the type declaration for svgs are missing.
Reference : https://stackoverflow.com/questions/44717164/unable-to-import-svg-files-in-typescript
For now, I just implemented the type declaration file and configured it in tsconfig.json as this article introduced, 
but if there's better way to do this, please let me know

**Pull request merged**